xbmc/utils/URIUtils.cpp

```diff
@@ -211,6 +211,17 @@ void URIUtils::Split(const std::string& strFileNameAndPath,
   strPath = strFileNameAndPath.substr(0, i+1);
   // everything to the right of the directory separator
   strFileName = strFileNameAndPath.substr(i+1);
+
+  // ignore options
+  i = strFileName.size() - 1;
+  while (i > 0)
+  {
+    char ch = strFileName[i];
+    if (ch == '?') break;
+    else i--;
+  }
+  if (i > 0)
+    strFileName = strFileName.substr(0, i);
```

Comment:
yeah, I think I was confused by strFileName
```std::string& strFileName``` is no pointer but a reference so it does get changed, correct?